### DISCUSSION TITLE
marketplace search can't find actions/checkout?

### DISCUSSION THREAD

**Discussion by ksully-execute** [community/community#54714] [Actions] [answered]
### Select Topic Area

Question

### Body

Newbie, why can't build-in actions like actions/cache and actions/checkout be found when you search in the marketplace?

**Comment by airtower-luna** [comment]
They probably simply aren't listed there. An action doesn't need to be listed in the marketplace to be usable, it's mostly a visibility thing. All that's needed is a public repository, e.g. `actions/checkout` at https://github.com/actions/checkout (mind how the action name simply is the path part of the URL). :slightly_smiling_face:

**Comment by ksully-execute** [comment]
But where is a list of all the actions and versions available?

**Comment by airtower-luna** [comment]
There simply is no list of all actions. You could try searching for `action.yml` and `action.yaml` files across all of GitHub, but I don't think the result would be very helpful. :sweat_smile: 

For versions, any tag, branch, and *commit ID* in an action repository is a valid version. Usually you should use one of the tagged commits, either by tag, or its commit ID.

**Comment by ksully-execute** [comment]
Perhaps I am confused then....so how does someone know what's actions exist that are not in the marketplace.....like actions/checkout or actions/cache...I see there are repos in github for the ones above, but surely there must be a list of them in docs somewhere? Telling you what version is the most recent, etc?

**Reply by jsoref** [reply]
There's no requirement that you know about all possible actions.

You might discover actions when you browse someone else's repository and see a workflow using an action.

There's no requirement that action versions be clearly documented.

Conventions dictate that repositories consider having a vaguely useful `README.md` and possibly use the `/releases` entrypoint, but neither are required.

If you can't figure out a given action / how it works / how it's maintained, then you probably don't want to use it.

**Comment by airtower-luna** [comment]
> I see there are repos in github for the ones above, but surely there must be a list of them in docs somewhere?

No, there simply isn't. I'm not sure how else to phrase that. :sweat_smile: The docs mention some core ones (like the ones you named, and a few more). Some projects & companies that provide actions for their tools and services might mention them in *their* docs. Some actions get published to the marketplace, some don't. It all depends on people who develop the actions.

Mind that *a lot of* actions aren't provided by GitHub, [anyone can make their own](https://docs.github.com/en/actions/creating-actions/about-custom-actions). Maybe that's where the confusion comes from?

> Telling you what version is the most recent, etc?

For versions look at the tags/releases in the repositories of the actions you use. If you just want to keep them up to date, consider [using Dependabot](https://docs.github.com/en/code-security/dependabot/working-with-dependabot/keeping-your-actions-up-to-date-with-dependabot) for that.

**Comment by ksully-execute** [comment]
Ok, thanks, close the case

**Comment by ksully-execute** [comment]
And just a comment, cheaping out and having ‘the community’ provide commonly needed actions is cheap. Secure companies will be vary wary of these things, who knows what it does?

By Github not stepping up and making common actions available, you’re risking your customers projects. The cost of reviewing the code for all marketplace items that are needed will cost a lot of time and money, pushing people to other products 


Ok, thanks, close the case

**Comment by airtower-luna** [comment]
You don't have to like how GitHub is doing things, but yelling at *a fellow user* who's just explaining how things *are* is extremely rude.

**Reply by ksully-execute** [reply]
that wasn't yelling, just making a comment, that's why I said 'just a comment'

**Reply by airtower-luna** [reply]
You're acting as if I was GitHub staff and as if I was responsible for their policies. I am not, and never was.

**Reply by ksully-execute** [reply]
my apologies, the comment certainly wasn't directed at you, it was just a comment meant for github itself. I appreciate your help.

**Comment by ksully-execute** [comment]
Thanks

**Comment by MikeMcC399** [comment, marked as answer]
@ksully-execute 

> why can't build-in actions like actions/cache and actions/checkout be found when you search in the marketplace?

I'm not sure why you couldn't find them, as they are both listed on https://github.com/marketplace. See

- https://github.com/marketplace/actions/cache
- https://github.com/marketplace/actions/checkout

**Comment by ksully-execute** [comment]
Ah, had not found that, was trying to find them in the marketplace search in the integrated editor while editing my workload. Thanks!

**Reply by MikeMcC399** [reply]
@ksully-execute 

I don't find the Marketplace search so easy to use, even when I'm searching for an action I know is in there, so somehow I'm not so surprised if you had difficulty! Maybe you'd like to mark my other reply as answer, so that others may find it?
